src/PFire.Core/Protocol/Messages/Bidirectional/ChatMessage.cs

```diff
@@ -24,6 +24,12 @@ public sealed class ChatMessage : XFireMessage
         public override void Process(XFireClient context)
         {
             var otherSession = context.Server.GetSession(SessionId);
+
+            if (otherSession == null)
```

Comment:
I think a good flow would be if `.GetSession(SessionId)` fails to find a session, we throw a `SessionNotFoundException` and handle it gracefully outside of any `XFireMessage` class. What do you think?

Reply:
The general rule with exceptions is not to use them for program flow control, I feel that returning null is the right thing to do here, IMHO; because as we can see from the "friends that are offline" fix, it would be easier to respond to null rather than catch an exception each time.